src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -132,6 +132,18 @@ CHIP_ERROR AES_CCM_encrypt(const uint8_t * plaintext, size_t plaintext_length, c
     int result               = 1;
     const EVP_CIPHER * type  = nullptr;
 
+    const uint8_t * empty_plaintext = Uint8::from_const_char("");
```

Comment:
Could we put this into a `constexpr uint8_t *kEmptyPlainText` that is a global constant in this file instead? we use this more than once.

Should ciphertext also be empty or is nullptr ok when plaintext length is empty?

Reply:
nullptr is ok when plaintext length is 0. The new unit tests ensure that this works across mbedTLS and OpenSSL implementations.